### DISCUSSION TITLE
FastAPI as python firebase functions - how to enable it

### DISCUSSION THREAD

**Discussion by smartrobos** [fastapi/fastapi#10382] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
below is how flask is enabled in Firebase Functions.
But don't see similar things in FastAPI

@https_fn.on_request()
def httpsflaskexample(req: https_fn.Request) -> https_fn.Response:
    with app.request_context(req.environ):
        return app.full_dispatch_request()
```

### Description

below is how flask is enabled in Firebase Functions.
But don't see similar things in FastAPI

### Operating System

macOS

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

3.11

**Comment by alta7700** [comment]
The same question.
Did you find a solution or did you use flask?

**Comment by sfahadco** [comment]
I am also looking for same solution.

**Comment by zammand** [comment]
Hi, I'm trying to deploy my app with firebase function, have you found a solution?

**Comment by unacceptable** [comment]
This guy seems to have made progress by defining an ASGI wrapper: https://www.reddit.com/r/Firebase/comments/1gw8zky/can_i_deploy_fastapi_code_in_firebase_functions/

I was able to get it running as well. 
<img width="2027" height="1368" alt="Screenshot 2025-07-14 at 11 26 18 PM" src="https://github.com/user-attachments/assets/f7d9f1a2-b210-42b7-b930-3c882a7c3717" />
I will see if I can refactor and simplify that a bit tomorrow.

In the mean time a gist of [No-Improvement4294](https://www.reddit.com/user/No-Improvement4294/)'s code that I modified can be found here: https://gist.github.com/unacceptable/747e1cb9d1fbe450d3b97e8c8ee1dc3d
